Python/Product/TestAdapter/TestContainerDiscoverer.cs

```diff
@@ -188,8 +190,15 @@ class TestContainerDiscoverer : ITestContainerDiscoverer, IDisposable {
         public event EventHandler TestContainersUpdated;
 
         private void NotifyContainerChanged() {
+            try {
+                _deferredTestChangeNotification.Change(500, Timeout.Infinite);
+            } catch (ObjectDisposedException) {
+            }
+        }
+
+        private void OnDeferredTestChanged(object state) {
             // guard against triggering multiple updates during initial load until setup is complete
-            if (!_firstLoad && !_isRefresh) {
```

Comment:
this !_isRefresh guard was pretty shady. Before I had to call Notify but then set _isRefresh = true afterwards so that it wouldn't get ignored.   Now during the refresh SetupSolution() wont trigger notifications so we dont need to guard this